cache_pdks/action.yml

```diff
@@ -0,0 +1,49 @@
+name: Install PDKs.
+
+inputs:
+  pdk_root:
+    description: PDK root to install files.
+    required: true
+  ol_path:
+    description: OpenLane path.
+    default: .
+  ol_tag:
+    description: OpenLane docker tag.
+    default: latest
+
+outputs:
+  cache_hash:
+    description: Hash for generated cache
+    value: ${{ steps.hash-generator.outputs.cache_hash }}
+
+runs:
+  using: composite
+  steps:
+
+    - name: Get PDKs cache hash.
+      id: hash-generator
+      shell: bash
+      run: |
+        OPEN_PDKS_COMMIT=$(python3 ${{ inputs.ol_path }}/dependencies/tool.py open_pdks -f commit)
+        SKYWATER_COMMIT=$(python3 ${{ inputs.ol_path }}/dependencies/tool.py sky130 -f commit)
+        MAGIC_COMMIT=$(python3 ${{ inputs.ol_path }}/dependencies/tool.py magic -f commit)
+        CACHE_HASH="open_pdks-$OPEN_PDKS_COMMIT-sky130-$SKYWATER_COMMIT-magic-$MAGIC_COMMIT"
+        echo "CACHE_HASH=$CACHE_HASH" >> $GITHUB_ENV
+        echo "::set-output name=cache_hash::${{ env.CACHE_HASH }}"
+
+    - name: Check for PDKs cache
+      uses: actions/cache@v3
+      id: cache
+      with:
+        path: ${{ runner.temp }}/pdks
+        key: ${{ env.CACHE_HASH }}
+
+    - name: Install PDKs
+      if: steps.cache.outputs.cache-hit != 'true'
+      shell: bash
+      run: |
+        make -C ${{ inputs.ol_path }} \
```

Comment:
any reason we need to use openlane here instead of simply building the open_pdks repo? (could be way faster since you wouldn't have to fetch the openlane container image)?

Reply:
See above.